### PR TITLE
[BPF] icmpv6 related works

### PR DESCRIPTION
**Pull request by tomastigera**
## Description
    
    Inner and outter headers are NATed and icmpv6 csum fixed up.
    
    Related ICMP follows conntrack and/or policy.

refs https://github.com/projectcalico/calico/issues/4736

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf-gpl/rpf.h [hunk 1]
```diff
@@ -83,15 +83,19 @@ static CALI_BPF_INLINE bool hep_rpf_check(struct cali_tc_ctx *ctx)
 			if (strict) {
 				ret = ctx->skb->ingress_ifindex == fib_params.ifindex;
 #ifdef IPVER6
+#ifdef VERIFIER_IS_COOL
```
Comment (fasaxc) [inline felix/bpf-gpl/rpf.h:86]: :rofl:

File: felix/bpf-gpl/tc.c [hunk 2]
```diff
@@ -1431,10 +1432,26 @@ static CALI_BPF_INLINE struct fwd calico_tc_skb_accepted(struct cali_tc_ctx *ctx
...
+				csum = bpf_csum_diff((__u32*)&STATE->ip_dst, sizeof(ipv6_addr_t),
+						     (__u32*)&STATE->ct_result.nat_sip, sizeof(ipv6_addr_t),
+						     csum);
+				int res = bpf_l4_csum_replace(ctx->skb, l4_csum_off, 0, csum,  BPF_F_PSEUDO_HDR);
+				if (res) {
+					deny_reason(ctx, CALI_REASON_CSUM_FAIL);
+					goto deny;
+				}
+#else
+				/* ... then fix the outer header IP first */
```
Comment (fasaxc) [inline felix/bpf-gpl/tc.c:1454]: first?

**Review by fasaxc — CHANGES_REQUESTED** [review]

File: felix/bpf-gpl/tc.c (hunk 2 above)
Comment (tomastigera) [reply felix/bpf-gpl/tc.c:1454]: This is original ipv4 code/comment, we do nat, we need to first fix the outter header before we move to the inner header in the rest of the code. But the comment should move back forward.

**Review by fasaxc — APPROVED** [review]

**Comment by tomastigera** [comment]
Only an unrelated flake

**Pull request merged**